daemon/cmd/endpoint.go

```diff
@@ -456,6 +457,15 @@ func (d *Daemon) createEndpoint(ctx context.Context, owner regeneration.Owner, e
 				}).Warningf("Endpoint has %s annotation, but BPF bandwidth manager is disabled. This annotation is ignored.",
 					bandwidth.EgressBandwidth)
 			}
+			if hwAddr, ok := annotations["cni.cilium.io/macAddr"]; ok {
+				m, err := mac.ParseMAC(hwAddr)
+				if err != nil {
+					log.WithField(logfields.K8sPodName, epTemplate.K8sNamespace+"/"+epTemplate.K8sPodName).
+						WithError(err).Error("Unable to parse MAC address")
+					return invalidDataError(ep, err)
+				}
+				ep.SetMac(m)
```

Comment:
With https://github.com/cilium/cilium/pull/26894/ we actually support more than one endpoint per pod. While lots of the logic to support multi-network is not yet upstreamed, it is clear that this is a use-case we are going to support.

Thus we need to think about the meaning of the pod annotation in case there are multiple endpoints per pod. 
Every endpoint will have its own interface, so we don't want to set the mac address the same for all interfaces. 

To be forward-compatible, I think we should check `DisableLegacyIdentifiers==false` before parsing the annotation and setting the MAC address here. By definition, only a single endpoint can have `DisableLegacyIdentifiers=false`, so that way we ensure that only one endpoint per pod has this custom MAC address set. This is also compatible with the current version of the CNI plugin which creates a single endpoint with `DisableLegacyIdentifiers=false`.

Reply:
Long-term, we also need to answer the question: How should the annotation look like if the pod has multiple interfaces? My above proposal basically suggests to only use the MAC for the "primary" endpoint, but a truly multi-network aware implementation might want to be able to define one MAC per interface.

Reply:
> With https://github.com/cilium/cilium/pull/26894 we actually support more than one endpoint per pod. 

I haven't seen this feature before. Thank you for your review.


> To be forward-compatible, I think we should check DisableLegacyIdentifiers==false before parsing the annotation and setting the MAC address here.

Got it.

> How should the annotation look like if the pod has multiple interfaces? My above proposal basically suggests to only use the MAC for the "primary"


So, do we define annotation as this 

```
cni.cilium.io/mac-address: ["e2:9c:30:38:52:62", "e2:9c:30:38:52:63",]
```

or like this



``` 
cni.cilium.io/primary-mac-address: "e2:9c:30:38:52:61"
cni.cilium.io/secondary-mac-address: ["e2:9c:30:38:52:62"]
``` 


I feel the second option is more expandable, what do you think?

Reply:
>. For now I think it's fine if we start with a solution that just looks at DisableLegacyIdentifiers and only sets the mac if DisableLegacyIdentifiers==false. 

Done.